redaxo/src/core/lib/console/setup/run.php

```diff
@@ -0,0 +1,550 @@
+                'MySQL Host',
+                'db-host',
+                $config['db'][1]['host'],
+                'Using MySQL Host "%s"',
+                $requiredValue
+            );
+            $dbLogin = $this->getOptionOrAsk(
+                'Login',
+                'db-login',
+                $config['db'][1]['login'],
+                'Using database login "%s"',
+                $requiredValue
+            );
+
+            $q = new Question('Password');
+            $q->setHidden(true);
+
+            $dbPassword = $this->getOptionOrAsk(
+                $q,
+                'db-password',
+                '',
+                'Using database password *secret*',
+                null
+            );
+
+            $dbName = $this->getOptionOrAsk(
+                'Database name',
+                'db-name',
+                $config['db'][1]['name'],
+                'Using database name "%s"',
+                $requiredValue
+            );
+
+            $dbCreate = $this->getOptionOrAsk(
+                new ConfirmationQuestion('Create database?', false),
+                'db-createdb',
+                false,
+                null,
+                static function ($value) {
+                    if (!in_array($value, ['yes', 'no', 'true', 'false'], true)) {
+                        throw new InvalidArgumentException('Unknown value "'.$value.'" specified');
+                    }
+                    return $value;
+                }
+            );
+
+            if (is_string($dbCreate)) {
+                $dbCreate = 'yes' === $dbCreate || 'true' === $dbCreate;
+                $io->success('Database will '.(!$dbCreate ? 'not ' : '').'be created');
+            }
+
+            $config['db'][1]['host'] = $dbHost;
+            $config['db'][1]['login'] = $dbLogin;
+            $config['db'][1]['password'] = $dbPassword;
+            $config['db'][1]['name'] = $dbName;
+
+            rex::setProperty('db', $config['db']);
+            try {
+                $err = rex_setup::checkDb($config, $dbCreate);
+            } catch (PDOException $e) {
+                $err = 'The following error occured: ' . $e->getMessage();
+            }
+
+            if ('' !== $err) {
+                $io->error($err);
+                if (!$input->isInteractive()) {
+                    return 1;
+                }
+                $this->forceAsking = true;
+            }
+        } while ('' !== $err);
+        $this->forceAsking = false;
+
+        $io->success('Database connection successfully established');
+
+        // ---------------------------------- step 5 . create db / demo
+        $io->title('Step 5 of 6 / Database');
+
+        $sql = rex_sql::factory();
+        $io->block('Database version: '.$sql->getDbType(). ' '.$sql->getDbVersion());
+
+        // Search for exports
+        $backups = [];
+        foreach (rex_backup::getBackupFiles('') as $file) {
+            if ('.sql' != substr($file, strlen($file) - 4)) {
```

Comment:
Mich wundert warum wir zuerst alle files abfragen und dann in der schleife  != sql abfragen und nicht gleich einen .tar.gz filter verwendet haben rex_backup::getBackupFiles('.tar.gz')